comp/core/workloadmeta/collectors/internal/crio/mock.go

```diff
@@ -0,0 +1,110 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build crio
+
+package crio
+
+import (
+	"context"
+	"errors"
+
+	v1 "k8s.io/cri-api/pkg/apis/runtime/v1"
+
+	workloadmeta "github.com/DataDog/datadog-agent/comp/core/workloadmeta/def"
+)
+
+// Helper functions to create pointer values for testing
+func floatPtr(f float64) *float64 {
+	return &f
+}
+
+func uintPtr(u uint64) *uint64 {
+	return &u
+}
+
+func intPtr(i int64) *int64 {
+	return &i
+}
```

Comment:
I found that a similar function using generic is already available in https://github.com/DataDog/datadog-agent/blob/main/pkg/util/pointer/pointer.go